src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -1245,13 +1256,21 @@ def get_statistic_points(
                     ]
                     all_weight_dims.extend(weight_dims)
 
-                # by default, reduce activations across all but the last dimension. The last dimension is
-                # assumed to be the hidden size dimension.
+                # Reduce activations across all but the hidden dimension.
                 n_dims = len(graph.get_output_edges_by_port_id(node, output_port_id)[0].tensor_shape)
-                reduction_axes = tuple(range(n_dims - 1))
+                # negative axis (e.g. -1 for the last axis) is converted into corresponding positive value
+                input_channel_axis = input_channel_axis % n_dims
+                reduction_axes = tuple(i for i in range(n_dims) if i != input_channel_axis)
+
+                if any(weight_dim > 3 for weight_dim in all_weight_dims):
+                    max_val = max(weight_dim for weight_dim in all_weight_dims)
+                    msg = f"Compression with {max_val} dimentional weight is not supported"
```

Comment:
we always keep convolutions with [4D weights](https://github.com/pytorch/pytorch/blob/v2.9.1/torch/nn/modules/conv.py#L168) in backup mode